### PR TITLE
Test coverage for issue42857

### PR DESCRIPTION
**Pull request by labkey-chrisj**
#### Rationale
One of our clients wants to bulk-import samples using a nameExpression that can resolve existing values via a lookup to another domain.  This test attempts to do that, and in the attempt appears to have found an [adjacent issue ](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=43394)

The test creates an integer-keyed list called Colors, with string columns ColorName and Code
values for ColorName and Code look like 'red' and 'rd', respectively.

It then creates a sampleType with a column named 'Color' that looks up the 'Colors' list (in the same project), and gives the sampleType a nameExpression with value 'TEST-${Color/Code}.
Via the bulk insert UI, it attempts to paste in a new Sample with value in the 'Color' column matching a value in the list's 'Code' column.

#### Related Pull Requests
n/a

#### Changes
just the new test case

### CODE REVIEW

File: src/org/labkey/test/tests/SampleTypeNameExpressionTest.java [hunk 1]
```diff
@@ -130,6 +135,54 @@ public void testParentAliasExpression()
...
+        String pasteData = "ColorLookup\tNoun\n" +
+                "red\tryder\n" +
+                "green\tgiant\n" +
+                "blue\tangel\n" +
+                "yellow\tjersey";
+
+        // now create a sampleType with a Color column that looks up to Colors
+        var sampleTypeDef = new SampleTypeDefinition(nameExpSamples)
+                .setFields(List.of(new FieldDefinition("ColorLookup", colorsLookup).setRangeURI(FieldDefinition.ColumnType.Integer.getRangeURI()),
+                        new FieldDefinition("Noun", FieldDefinition.ColumnType.String)))
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/SampleTypeNameExpressionTest.java]: The correct pattern to set the type of a lookup is to call `setTableType`. This `setRangeURI` isn't necessary.

File: src/org/labkey/test/tests/SampleTypeNameExpressionTest.java [hunk 2]
```diff
@@ -130,6 +135,54 @@ public void testParentAliasExpression()
                 "Parent", "Jessi");
     }
 
+    // Issue 42857: samples: bulk import with name expression containing lookup fails to convert lookup string value
+    @Test
+    public void testLookupNameExpression() throws Exception
+    {
+        String lookupList = "Colors";
+        FieldDefinition.LookupInfo colorsLookup = new FieldDefinition.LookupInfo(getProjectName(), "lists", lookupList);
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/SampleTypeNameExpressionTest.java]:
```suggestion
        FieldDefinition.LookupInfo colorsLookup = new FieldDefinition.LookupInfo(getProjectName(), "lists", lookupList).setTableType(FieldDefinition.ColumnType.Integer);
```

File: src/org/labkey/test/params/FieldDefinition.java [hunk 3]
```diff
@@ -111,6 +111,13 @@ public FieldDefinition setFormat(String format)
         return this;
     }
 
+    @Override
+    public FieldDefinition setRangeURI(String rangeURI)
+    {
+        super.setRangeURI(rangeURI);
+        return this;
+    }
+
```
Comment (labkey-tchad) [inline src/org/labkey/test/params/FieldDefinition.java]: Don't need this now, right?
```suggestion
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

File: src/org/labkey/test/params/FieldDefinition.java (hunk 3 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/params/FieldDefinition.java]: Let's hope you're right

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**